Andy/feature/table ready

AndrewCerveny opened the pull request:
# Description

Has an added function inspired by Brett who found a way to compare the ingredients of a panty to the ingredients of a recipe. Using this function we create an object with 2 keys. [ notReadyToCook] and [readyTooCook] 
because our thumbnails are dynamically added. Brett and I rubber ducked a way to get the function to also appear dynamically. By adding if statements that say if current page is this and our ready to cook includes ( recipe) then sow this type of html. I added two css class rules that will change the opacity of the image and tell the user which recipes are ready to be cooked or not based on pantry items. 

1. we might want to trigger the pantry items subtraction only inside a certain condition. where we invoke it in this if statement or we should create a fuction that says ( if recipe is part of the ready to cook values and is clicked then we subtract.
2. also do we want to incorporate units of measurement for ingredients? 

  

Fixes # (issue)
A way to add the the user story and tell a user if they have enough ingredients in their pantry to cover a recipe or not. 
## Type of change
- [ ] HTML
- [x] CSS
- [x] DOM
- [ ] DataModel
- [ ] API
- [x] ScriptsJS

Please delete options that are not relevant.


- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

I have saved multiple recipes and altered the pantry amounts using the input to tailer to the recipe needs. the Options of not enough to cook and ready to cook show with their images  varying in opacity. 


- [x] Test Approved


# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

areas affected: 
scripts
line 588 added a function sortByCookable(currentUser) dynamic to whatever pantry is given on pageload
Line 284 DisplayRecipeThumbnails() this is where we invoke our function and display the different paths
// in each html structure there are class changes that change their state on upload. 

styles.css
281. transparent class which was added to each image making them more transparent
285 Meal ready this is added to an added P tag that will either tell the user 1. not enough Ingredients 2. Ready to Cook!

AndrewCerveny committed:
add a sortMeal funtion that will take a currentUser as arguement

AndrewCerveny committed:
make target thumbnail more readible for code placement.

AndrewCerveny committed:
add a warning message to not ready ingredients

AndrewCerveny committed:
Adds a ready to cook message for users

AndrewCerveny committed:
add a user story of having a prep able meal

AndrewCerveny committed:
add notification when recipes are ready to cook

AndrewCerveny committed:
check over functionality of recipestocook

AndrewCerveny committed:
remove an extra console.log in funciton

bkuhn2 commented:
Looks awesome, the functionality works as expected - there was one bug not actually this new code though, deals with the event bubbling when clicking on a saved recipe thumbnail now that there is an additional "transparent" class there. Very easy fix, I tested it and all that needs to be done is:

- Within the loadSpecificRecipe() function (line 381), change the "event.target.className === "single-recipe-img"" on lines 382 and 393 to be "event.target.classList.contains("single-recipe-img"), pictured below
![Screen Shot 2022-11-04 at 5 56 34 PM](https://user-images.githubusercontent.com/110054994/200091604-678ba178-cb0a-4cd5-9bfd-c79a434af531.png)

AndrewCerveny committed:
Add changes to seperate class clashing for thumbnail images

AndrewCerveny commented:
scripts line 381  loadSpecificRecipe(event)
 I updated line 382 to  event.target.classList.contains( " single-recipe-img")
I also updated like 393 ( to the same thing but rapped the contains around the whole if block ending on 395 because is housed an <or<|| > conditional.

bkuhn2 commented:
That block actually was already contained in a parentheses, looks like another bug popped up - I can show you on zoom

AndrewCerveny committed:
fixed the wrapping on line 393 scripts
